Source/stores.cpp

```diff
@@ -2275,20 +2272,19 @@ BOOL StoreGoldFit(int idx)
 
 	for (i = 0; i < plr[myplr]._pNumInv; i++) {
 		if (plr[myplr].InvList[i]._itype == ITYPE_GOLD && plr[myplr].InvList[i]._ivalue != 5000) {
-			cost += plr[myplr].InvList[i]._ivalue;
-			if (cost > 5000)
-				cost -= 5000;
-			else
+			if (cost + plr[myplr].InvList[i]._ivalue <= 5000)
 				cost = 0;
+			else
+				cost -= 5000 - plr[myplr].InvList[i]._ivalue;

```

Comment:
Maybe it's just me, but for the sake of clarity I'd rewrite it so:
`cost = cost + plr[myplr].InvList[i]._ivalue - 5000;`
... or just use `+=` here.

Reply:
OTOH (after trying to understand the logic rather than simplify math), maybe original looks really better.